OpenRA.Mods.Common/Traits/Harvester.cs

```diff
@@ -323,11 +326,15 @@ public bool TickUnload(Actor self, Actor proc)
 			{
 				var type = contents.First().Key;
 				var iao = proc.Trait<IAcceptResources>();
-				if (!iao.CanGiveResource(type.ValuePerUnit))
+				var count = contents[type] - Info.BaleUnloadAmount < 0 ? contents[type] : Info.BaleUnloadAmount;
```

Comment:
This can simplify to `var count = Math.Min(contents[type], Info.BaleUnloadAmount);`